inventory_management_system_api/routers/v1/unit.py

```diff
@@ -6,17 +6,65 @@
 import logging
 from typing import Annotated
 
-from fastapi import APIRouter, Depends
+from fastapi import APIRouter, Depends, status, HTTPException, Path
 
-from inventory_management_system_api.schemas.unit import UnitSchema
+from inventory_management_system_api.core.exceptions import DuplicateRecordError, InvalidObjectIdError
+from inventory_management_system_api.schemas.unit import UnitPostRequestSchema, UnitSchema
 from inventory_management_system_api.services.unit import UnitService
 
 logger = logging.getLogger()
 
 router = APIRouter(prefix="/v1/units", tags=["units"])
 
 
-@router.get(path="", summary="Get Units", response_description="List of Units")
+@router.post(
+    path="",
+    summary="Create a new unit",
+    response_description="The created unit",
+    status_code=status.HTTP_201_CREATED,
+)
+def create_unit(
+    unit: UnitPostRequestSchema,
+    unit_service: Annotated[UnitService, Depends(UnitService)],
+) -> UnitSchema:
+    # pylint: disable=missing-function-docstring
+    logger.info("Creating a new unit")
+    logger.debug("Unit data: %s", unit)
+
+    try:
+        unit = unit_service.create(unit)
+        return UnitSchema(**unit.model_dump())
+
+    except DuplicateRecordError as exc:
+        message = "A unit with the same value already exists"
+        logger.exception(message)
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=message) from exc
+
+
+@router.get(
+    path="/{unit_id}",
+    summary="Get a unit by ID",
+    response_description="Single unit",
+)
+def get_unit(
+    unit_id: Annotated[str, Path(description="The ID of the unit to be retrieved")],
+    unit_service: Annotated[UnitService, Depends(UnitService)],
+) -> UnitSchema:
+    # pylint: disable=missing-function-docstring
+    logger.info("Getting unit with ID %s", unit_id)
+    message = "Unit not found"
+    try:
+        unit = unit_service.get(unit_id)
+        if not unit:
+            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=message)
+    except InvalidObjectIdError as exc:
+        logger.exception("The ID is not a valid ObjectId value")
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=message) from exc
+
+    return UnitSchema(**unit.model_dump())
+
+
+@router.get(path="", summary="Get units", response_description="List of units")
```

Comment:
```suggestion
@router.get(path="", summary="Get Units", response_description="List of Units")
```
Very minor point but I intentionally used these to distinguish a `Unit` object and a `unit` itself. Its the same above.